lib/enumerations/value.rb

```diff
@@ -54,7 +54,7 @@ def create_instance_methods
     #
     def define_attributes_getters
       @attributes.each do |key, _|
-        next if respond_to?(key)
+        next if self.class.instance_methods(false).include?(key)
```

Comment:
This one is a bit tricky as it would only fail with the key `:name`.

In older Rubies, `#name` isn't defined so `respond_to?(:name)` returns false. In Ruby 3 however, some superclass of `self.class` implements `#name` and that would conflict with a user-defined attribute `name` because `respond_to?(:name)` would return true and when `#name` would be called on an enumeration value, it would call the wrong method.

Since we only need to check for definitions of instance methods from this class (and not all ancestors), I refactored the logic so it doesn't search all ancestor classes.